sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/StorageImplUtils.java

```diff
@@ -297,4 +301,53 @@ public static void copyToOutputStream(InputStream source, long writeLength, Outp
             count = source.read(retrievedBuff, 0, nextCopy);
         }
     }
+
+    /**
+     * Logs the string to sign if a valid context is provided.
+     *
+     * @param logger {@link ClientLogger}
+     * @param stringToSign The string to sign to log.
+     * @param context Additional context to determine if the string to sign should be logged.
+     */
+    public static void logStringToSign(ClientLogger logger, String stringToSign, Context context) {
+        if (context != null && Boolean.TRUE.equals(context.getData(Constants.STORAGE_LOG_STRING_TO_SIGN).orElse(false))) {
+            logger.info("The string to sign computed by the SDK is: {}{}", stringToSign,
+                System.lineSeparator());
+            logger.warning("Please remember to disable '{}' before going to production as this "
```

Comment:
Let's make this log message a constant string as it doesn't have any instance data